nntrainer/include/layer.h

```diff
@@ -128,16 +128,16 @@ class Layer {
    * @param[in] in List of Input Tensors taken by this layer
    * @retval    List of Output Tensors
    */
-  virtual sharedConstTensor forwarding(sharedConstTensor in) = 0;
+  virtual sharedConstTensors forwarding(sharedConstTensors in) = 0;
```

Comment:
please also update weight_list from shared_ptr<Weight> to vector<Weight>.
It is supposed to follow the same semantics as tensor list.

Reply:
Well, I'm not sure. Basic rule here is that layer only has one input except several layer.

Reply:
shared_ptr<Weight> is representing a fixed size array of weights with the number elements in array stored separately.
So shared_ptr<Weight> can be replaced with vector<Weight>.

Reply:
In order to do this, there are a lot more changes. I will do it later pr.

Reply:
Sure.